File: j2k/src/org/jetbrains/jet/j2k/ast/TypeCastExpression.java
package org.jetbrains.jet.j2k.ast;

import org.jetbrains.annotations.NotNull;

/**
 * @author ignatov
 */
public class TypeCastExpression extends Expression {
  private final Type myType;
  private final Expression myExpression;

  public TypeCastExpression(Type type, Expression expression) {
    myType = type;
    myExpression = expression;
  }

  @NotNull
  @Override
  public String toKotlin() {
    return "(" + myExpression.toKotlin() + SPACE + "as" + SPACE + myType.toKotlin() + ")";
  }
}

File: src/org/jetbrains/jet/j2k/ast/TypeCastExpression.kt
package org.jetbrains.jet.j2k.ast

import org.jetbrains.jet.j2k.ast.types.Type

public open class TypeCastExpression(val `type` : Type, val expression : Expression) : Expression() {
    public override fun toKotlin() = "(" + expression.toKotlin() + " as " + `type`.toKotlin() + ")"
}
